sdks/python/apache_beam/runners/common.py

```diff
@@ -761,6 +761,17 @@ def __init__(self,
       self.current_window_index = None
       self.stop_window_index = None
 
+    # TODO(https://github.com/apache/beam/issues/28776): Remove caching after
+    # fully rolling out.
+    # If true, always recalculate window args. If false, has_cached_window_args
+    # and has_cached_window_batch_args will be set to true if the corresponding
+    # self.args_for_process,have been updated and should be reused directly.
+    self.recalculate_window_args = (
+        self.has_windowed_inputs or 'disable_global_windowed_args_caching' in
+        RuntimeValueProvider.experiments)
```

Comment:
I haven't tried this method of fetching experiments; I recommend you try to test that it works on Dataflow. For example, you could throw a runtime error for testing purposes.

Reply:
I verified this with my refreshing side input test. It sees only one version without the experiment and sees multiple side input versions and passes with the experiment.

(I copied this from another experiment in this same file)